python/mcap/mcap/mcap0/reader.py

```diff
@@ -150,31 +155,40 @@ def iter_messages(
             timestamp are not included.
         :param end_time: an integer nanosecond timestamp. if provided, messages logged after this
             timestamp are not included.
+        :param log_time_order: if True, messages will be yielded in order of their log times. If
+            False, messages will be yielded in the order they appear in the MCAP file.
         """
         summary = self.get_summary()
-        assert summary is not None
         if summary is None:
             # no index available, use a non-seeking reader to read linearly through the stream.
             self._stream.seek(0, io.SEEK_SET)
             return NonSeekingReader(self._stream).iter_messages(
-                topics, start_time, end_time
+                topics, start_time, end_time, log_time_order
             )
+
+        message_queue = MessageQueue(log_time_order=log_time_order)
         for chunk_index in _chunks_matching_topics(
             summary, topics, start_time, end_time
         ):
-            self._stream.seek(chunk_index.chunk_start_offset + 1 + 8, io.SEEK_SET)
-            chunk = Chunk.read(ReadDataStream(self._stream))
-            for record in breakup_chunk(chunk):
-                if isinstance(record, Message):
-                    channel = summary.channels[record.channel_id]
-                    if topics is not None and channel.topic not in topics:
-                        continue
-                    if start_time is not None and record.log_time < start_time:
-                        continue
-                    if end_time is not None and record.log_time >= end_time:
-                        continue
-                    schema = summary.schemas[channel.schema_id]
-                    yield (schema, channel, record)
+            message_queue.push(chunk_index)
+        while message_queue.len() > 0:
```

Comment:
pythonically this should be `while message_queue:`

or perhaps in python3.8+, `while next_item := message_queue.pop():`